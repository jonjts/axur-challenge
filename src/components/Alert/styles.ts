import styled from 'styled-components';

export const Container = styled.div<{ color?: string }>`
  max-width: 80%;
  width: 400px;
  height: 100px;
  display: flex;
  flex: 1;
  flex-direction: column;
  padding: ${({ theme: { spacings: { sm, lg } } }) => `${lg.pixel} ${sm.pixel} ${sm.pixel} ${lg.pixel}`};
  background-color: ${({ color, theme: { colors } }) => color || colors.danger};
  border-radius: ${({ theme: { border } }) => border};
  z-index: 1;
  position: absolute;
  top: 8%;
  right: 10%;
  animation-name: alert-incoming;
  animation-duration: 0.8s;

  h2 {
      margin: 0%;
      color: #FFF;
  }

  p {
      color: #FFF;
  }
`;
