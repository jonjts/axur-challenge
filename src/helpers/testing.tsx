import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import ThemeProvider from '../providers/Theme';
import InspectionProvider from '../providers/Inspection'

export const putProviders = (component: JSX.Element) => (
    <ThemeProvider>
        <InspectionProvider>
            {component}
        </InspectionProvider>
    </ThemeProvider>
);

export const matchSnapshotWithProviders = (component: JSX.Element) => {
    const tree = renderer.create(putProviders(component)).toJSON();
    expect(tree).toMatchSnapshot();
    return tree;
};

export const renderWithProviders = (component: JSX.Element) => {
    return render(putProviders(component));
};
