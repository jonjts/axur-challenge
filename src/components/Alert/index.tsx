import React from 'react';

import { Container } from './styles';

type Props = {
    title?: string,
    description?: string,
    color?: string
};

const Alert: React.FC<Props> = ({ title, description, color }) => {
    return (
        <Container color={color}>
            <h2>
                {title}
            </h2>
            <p>
                {description}
            </p>
        </Container>
    );
};

export default Alert;