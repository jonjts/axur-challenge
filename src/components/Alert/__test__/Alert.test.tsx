import { matchSnapshotWithProviders, renderWithProviders } from '../../../helpers/testing';

import Alert from '../index';

describe('Alert', () => {
  test('match with snapshot', () => {
    matchSnapshotWithProviders(<Alert />);
  });

  test('renders correctly', () => {
    renderWithProviders(<Alert />);
  });
});
