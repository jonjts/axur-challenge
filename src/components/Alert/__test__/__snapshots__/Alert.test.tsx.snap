// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert match with snapshot 1`] = `
<div
  className="sc-bdfBwQ dWpYVM"
>
  <h2 />
  <p />
</div>
`;
